lib/Dialect/RTL/RTLDialect.cpp

```diff
@@ -16,16 +16,69 @@
 #include "mlir/IR/Builders.h"
 #include "mlir/IR/BuiltinTypes.h"
 #include "mlir/IR/DialectImplementation.h"
+#include "mlir/Transforms/InliningUtils.h"
 
 using namespace circt;
 using namespace rtl;
 
 //===----------------------------------------------------------------------===//
-// Dialect specification.
+// Inliner Dialect Interface
 //===----------------------------------------------------------------------===//
 
 namespace {
+struct RTLInlinerInterface : public mlir::DialectInlinerInterface {
+  using mlir::DialectInlinerInterface::DialectInlinerInterface;
+
+  /// Returns true is the `callable` operation can be inlined into the location
+  /// of the `call` operation.
+  bool isLegalToInline(Operation *call, Operation *callable,
+                       bool wouldBeCloned) const override final {
+    // Only inline an  operation which is marked inline
+    return call->hasAttr("inline");
```

Comment:
How does this interface and the general inlining pass decide to inline?  This seems to be mixing the legality with the profitability.

Reply:
Yeah, I think that this is conflating "can this inline" with "should this inline". I think we should remove this check, since it is always legal to inline an RTL module.